### PR TITLE
[PDP] Make template notebooks (trying again)

### PR DESCRIPTION
**Pull request by bdewilde**
## changes

- adds a template notebook for (pdp) data assessment and eda
- adds a template notebook for (pdp) modeling dataset prep
- fixes a minor bug in code committed just recently, oops

## context

We want to give folks a clear blueprint to follow when doing this work, leaving plenty of space for artistic license and necessary customization.

## questions

- How's the template look? Anything notably absent or confusing??

### CODE REVIEW

**Review by kaylawilding — CHANGES_REQUESTED** [review]

File: notebooks/pdp/01-data-assessment-eda-TEMPLATE.py [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC # Read and Validate Raw Data
+
+# COMMAND ----------
+
+# configure where schools' raw data is stored in Unity Catalog
+catalog = "sst_dev"
+schema = "SCHOOL_bronze"
```
[the remark this answers is not in the document]
Comment (bdewilde) [reply notebooks/pdp/01-data-assessment-eda-TEMPLATE.py:82]: Yup, I did something like this in USCB. This notebook is just a draft, I was hoping you'd be updating it, but I can extend what I have further given recent experience if you'd like.

[the remark this answers is not in the document]
Comment (kaylawilding) [reply notebooks/pdp/01-data-assessment-eda-TEMPLATE.py:82]: Oh cool, yes I can update it totally, you can ignore my comments if you want and just merge and I can update it all!

File: notebooks/pdp/01-data-assessment-eda-TEMPLATE.py [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+
+# COMMAND ----------
+
+# read without any schema validation, so we can look at the data "raw"
+fname_course = "SCHOOL_COURSE_AR_DEID_DTTM.csv"
+fpath_course = os.path.join(path_volume, fname_course)
+df_course_raw = pdp.dataio.read_raw_pdp_course_data_from_file(
+    fpath_course, schema=None, dttm_format="%Y%m%d.0"
+)
+print(f"rows x cols = {df_course_raw.shape}")
```
[the remark this answers is not in the document]
Comment (bdewilde) [reply notebooks/pdp/01-data-assessment-eda-TEMPLATE.py:102]: Uniqueness checks are embedded within the raw data schemas -- this gets flagged automatically when validating the data 👍

File: notebooks/pdp/01-data-assessment-eda-TEMPLATE.py [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+# COMMAND ----------
+
+# try to read data while validating with the "base" PDP schema
+# TODO: update schema import path once https://github.com/datakind/student-success-tool/pull/19 is merged
+base_course_schema = pdp.schemas.base.RawPDPCourseDataSchema
+
+df_course = pdp.dataio.read_raw_pdp_course_data_from_file(
+    fpath_course, schema=base_course_schema, dttm_format="%Y%m%d.0"
+)
+df_course
```
[the remark this answers is not in the document]
Comment (bdewilde) [reply notebooks/pdp/01-data-assessment-eda-TEMPLATE.py:122]: This is just a template -- all are free to customize as they prefer! :)

File: notebooks/pdp/01-data-assessment-eda-TEMPLATE.py [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+# MAGIC - ...
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## null values
+
+# COMMAND ----------
+
+_ = msno.matrix(
```
[the remark this answers is not in the document]
Comment (bdewilde) [reply notebooks/pdp/01-data-assessment-eda-TEMPLATE.py:274]: Sorry, but I fully don't understand this comment. Why would variable names be changing? And why would that matter here?

**Comment by bdewilde** [comment]
Hey, I'm trying to do a v0.1 release of this package, and I'd strongly prefer to have template notebooks included. So, I'm going to merge this in, but let's plan to iterate on them for a follow-up release.

**Review by Eyong20 — COMMENTED** [review]
reviewed

**Pull request merged**